packages/synapse-react-client/src/components/EntityFinder/details/view/DetailsView.tsx

```diff
@@ -252,119 +505,190 @@ export function DetailsView({
+    ).length > 0
   )
 
-  const SelectAllCheckboxRenderer = useMemo(() => {
-    // Enabled if there's at least one visible & selectable entity, OR there's a page we haven't fetched
-    const isEnabled =
-      hasNextPage ||
-      entities.filter(
-        e =>
-          selectableTypes.includes(getEntityTypeFromHeader(e)) &&
-          visibleTypes.includes(getEntityTypeFromHeader(e)),
-      ).length > 0
-    return (
-      enableSelectAll && (
-        <div
-          data-testid="Select All"
-          style={isEnabled ? { cursor: 'pointer' } : { cursor: 'not-allowed' }}
-          onClick={() => {
-            if (isEnabled) {
-              setShouldSelectAll(true)
-            }
-          }}
-        >
-          <Checkbox
-            inputProps={{ 'aria-label': 'Select All' }}
-            checked={selectAllIsChecked}
-            disabled={!isEnabled}
-            onChange={() => {
-              // no-op
-            }}
-          />
-        </div>
-      )
+  const columns = useMemo(
+    () =>
+      getColumns({
+        setCurrentContainer,
+        isSelectAllVisible: isSelectAllVisible,
+        isSelectAllDisabled: isSelectAllDisabled,
+        isSelectAllChecked: selectAllIsChecked,
+        onSelectAll: () => {
+          setShouldSelectAll(true)
+        },
+        versionSelection,
+        toggleSelection,
+        sortableColumns,
+      }),
+    [
+      isSelectAllDisabled,
+      isSelectAllVisible,
+      selectAllIsChecked,
+      setCurrentContainer,
+      sortableColumns,
+      toggleSelection,
+      versionSelection,
+    ],
+  )
+
+  const hideColumnOverrides = useMemo(() => {
+    return hiddenColumns.reduce(
+      (acc: Partial<Record<DetailsViewColumn, boolean>>, column) => {
+        acc[column] = false
+        return acc
+      },
+      {},
     )
-  }, [
-    enableSelectAll,
-    entities,
-    hasNextPage,
-    selectAllIsChecked,
-    selectableTypes,
-    visibleTypes,
-  ])
+  }, [hiddenColumns])
+
+  const table: Table<EntityFinderTableViewRowData> =
+    useReactTable<EntityFinderTableViewRowData>({
+      data: tableData,
+      columns,
+      getCoreRowModel: getCoreRowModel(),
+      columnResizeMode: 'onChange',
+      manualSorting: true,
+      enableSorting,
+      enableMultiSort,
+      onSortingChange,
+      state: {
+        sorting,
+        columnVisibility: {
+          [DetailsViewColumn.SELECTED]: showSelectColumn,
+          [DetailsViewColumn.VERSION]:
+            versionSelection !== VersionSelectionType.DISALLOWED,
+          [DetailsViewColumn.SIZE]: visibleTypes.includes(EntityType.FILE),
+          [DetailsViewColumn.MD5]: visibleTypes.includes(EntityType.FILE),
+          [DetailsViewColumn.ADD_TO_DOWNLOAD_CART]: visibleTypes.includes(
+            EntityType.FILE,
+          ),
+          [DetailsViewColumn.DIRECT_DOWNLOAD]: visibleTypes.includes(
+            EntityType.FILE,
+          ),
+          ...hideColumnOverrides,
```

Comment:
Columns passed via `hiddenColumns` prop will pass along show/hide state to `useReactTable` here